go/vt/vtgate/vtgate.go

```diff
@@ -1046,14 +1025,25 @@ func annotateBoundShardQueriesAsUnfriendly(queries []*vtgatepb.BoundShardQuery)
 	}
 }
 
-// parseKeyspaceOptionalShard parses a "keyspace/shard" or "keyspace:shard" string
-// and extracts the parts. If a shard is not specified, it's
-// returned as empty string. We need to support : and / in vtgate because some clients
-// can't support our default of /. Everywhere else we only support /.
-func parseKeyspaceOptionalShard(keyspaceShard string) (string, string) {
-	last := strings.LastIndexAny(keyspaceShard, "/:")
-	if last == -1 {
-		return keyspaceShard, ""
+// parseTarget parses the string representation of a Target
+// of the form keyspace:shard@tablet_type. You can use a / instead of a :.
```

Comment:
I wonder if we should use the same separator for all parts, i.e. `keyspace:shard:tablet_type`. Might make it easier for people to remember which to use in which spot. Not stuck on that though, just a thought.